src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,183 @@
+                return ui.eventError();
+            }
+
+        } else if (text.startsWith("mark")) {
+            try {
+                if (text.equalsIgnoreCase("mark") || text.equalsIgnoreCase("mark ")) {
+                    return ui.noNumberToMark();
+                } else {
+                    if (Integer.parseInt(text.replace("mark ", "")) - 1 < storage.taskList.arrayList.size()
+                            && Integer.parseInt(text.replace("mark ", "")) > 0) {
+                        Task toBeMarked = storage.taskList.arrayList.get(Integer.parseInt(text.replace("mark ", "")) - 1);
+                        if (!toBeMarked.getIsDone()) {
+                            storage.taskList.mark(toBeMarked);
+                            return ui.mark(text);
+                        } else {
+                            return ui.alreadyMarked();
+                        }
+                    } else {
+                        return ui.noItemToMark();
+                    }
+                }
+            } catch (NumberFormatException error) {
+                return ui.markFormatError();
+            }
+
+        } else if (text.startsWith("unmark")) {
+            try {
+                if (text.equalsIgnoreCase("unmark") || text.equalsIgnoreCase("unmark ")) {
+                    return ui.noNumberToUnmark();
+                } else {
+                    if (Integer.parseInt(text.replace("unmark ", "")) - 1 < storage.taskList.arrayList.size()
+                            && Integer.parseInt(text.replace("unmark ", "")) > 0) {
+                        Task toBeUnmarked = storage.taskList.arrayList.get(Integer.parseInt(text.replace("unmark ", "")) - 1);
+                        if (toBeUnmarked.getIsDone()) {
+                            storage.taskList.unmark(toBeUnmarked);
+                            return ui.unmark(text);
+                        } else {
+                            return ui.alreadyUnmarked();
+                        }
+                    } else {
+                        return ui.noItemToUnmark();
+                    }
+                }
+            } catch (NumberFormatException error) {
+                return ui.unmarkFormatError();
+            }
+
+        } else if (text.startsWith("help")) {
+            return ui.help();
+
+        } else if (text.startsWith("find")) {
+            if (text.startsWith("find ")) {
+                String search = text.replace("find ", "");
+                return ui.find(search);
+            } else if (text.equalsIgnoreCase("find")) {
+                return ui.noSearchError();
+            } else {
+                return ui.findError();
+            }
+
+        } else {
+            try {
+                throw new DukeException.DukeCommandException("Invalid command");
+            } catch (DukeException.DukeCommandException error) {
+                return ui.generalError(error);
+            }
+        }
+    }
```

Comment:
Perhaps using a switch statement could be neater? Also, given how you have many specific Duke Exceptions, you could consider moving all the catch blocks below, and capture the rest of the logic in one try block.